pkg/daemon/ceph/osd/device.go

```diff
@@ -651,7 +651,7 @@ func addOSDToCrushMap(context *clusterd.Context, config *osdConfig, clusterName,
 	logger.Infof("adding %s (%s), bytes: %d, weight: %.4f, to crush map at '%s'",
 		osdEntity, osdDataPath, totalBytes, weight, location)
 	args := []string{"osd", "crush", "create-or-move", strconv.Itoa(osdID), fmt.Sprintf("%.4f", weight)}
-	args = append(args, strings.Split(location, " ")...)
```

Comment:
The CI is failing with this error in the osd prepare log (which I found by running locally since that log is missing in jenkins):
```
2019-05-16 22:43:38.338874 I | exec: Running command: ceph osd crush create-or-move 0 0.0166 root=default host=test-prefix-this-is-a-very-long-hostname-minikube --connect-timeout=15 --cluster=block-test-ns --conf=/var/lib/rook/block-test-ns/block-test-ns.config --keyring=/var/lib/rook/block-test-ns/client.admin.keyring --format json --out-file /tmp/553091963
2019-05-16 22:43:38.663767 I | exec: Invalid command: invalid chars   in root=default host=test-prefix-this-is-a-very-long-hostname-minikube
osd crush create-or-move <osdname (id|osd.id)> <float[0.0-]> <args> [<args>...] :  create entry or move existing entry for <name> <weight> at/to location <args>
Error EINVAL: invalid command
2019-05-16 22:43:38.664077 E | cephosd: failed to config osd in path /data/rook-test/block-test-ns/test-5577006791947779410. failed to initialize OSD at /data/rook-test/block-test-ns/test-5577006791947779410/osd0: failed adding osd.0 to crush map: exit status 22
```

There must still be somewhere in the code that the location is appended with a space, so the command is confused when the split does not happen based on the space. Either we need to find where that space is added or perhaps just support parsing based on both space and comma, which would ensure backward compatibility.

Reply:
Is that a upgrade of a cluster which had the location separated by " " already? I always booted up a fresh cluster...

Reply:
@travisn I found that line which also seperates a location by space: https://github.com/rook/rook/blob/0dacfd4a0cc1f9420e2aa3b7810f286d84c9b6d0/pkg/operator/k8sutil/node.go#L270

Do you have location labels on your nodes?

Reply:
ok i think i found the problem. Pipeline is partially running through, the other integration test problems are not related to this bugfix. https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/PR-3119/18/pipeline/